src/main/java/org/fcrepo/audit/InternalAuditor.java

```diff
@@ -139,22 +148,15 @@ public void recordEvent(final FedoraEvent event) {
         LOGGER.debug("Event detected: {} {}", event.getUserID(), event.getPath());
         boolean isParentNodeLastModifiedEvent = false;
         final String eventType = AuditUtils.getEventURIs(event.getTypes());
-        final Set<String> properties = event.getProperties();
-        if (eventType.contains(AuditProperties.PROPERTY_CHANGED)) {
-            isParentNodeLastModifiedEvent = true;
-            final Iterator<String> propertiesIter = properties.iterator();
-            String property;
-            while (properties.iterator().hasNext()) {
-                property = propertiesIter.next();
-                if (!property.equals(AuditProperties.LAST_MODIFIED) &&
-                        !property.equals(AuditProperties.LAST_MODIFIED_BY)) {
-                    /* adding/removing a file updates the lastModified property of the parent container,
-                    so ignore updates when only lastModified is changed */
-                    isParentNodeLastModifiedEvent = false;
-                    break;
-                }
-            }
+
+        /* adding/removing a file updates the lastModified property of the parent container,
+           so ignore updates when only lastModified is changed. */
+        if (eventType.contains(PROPERTY_CHANGED)) {
+            isParentNodeLastModifiedEvent = !event.getProperties().stream().filter(p ->
+                !p.equals(LAST_MODIFIED) && !p.equals(LAST_MODIFIED_BY)
+            ).findFirst().isPresent();
```

Comment:
Oh, shazam! I thought we actually needed the property itself or I would have suggested this at first! :)